recipes/glog/all/conanfile.py

```diff
@@ -18,83 +21,89 @@ class GlogConan(ConanFile):
         "fPIC": [True, False],
         "with_gflags": [True, False],
         "with_threads": [True, False],
+        "with_unwind": [True, False],
     }
     default_options = {
         "shared": False,
         "fPIC": True,
         "with_gflags": True,
         "with_threads": True,
+        "with_unwind": True,
     }
 
-    exports_sources = ["CMakeLists.txt", "patches/**"]
-    generators = "cmake", "cmake_find_package"
-    _cmake = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
+    def export_sources(self):
+        for p in self.conan_data.get("patches", {}).get(self.version, []):
+            copy(self, p["patch_file"], self.recipe_folder, self.export_sources_folder)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
+        if self.settings.os not in ["Linux", "FreeBSD"] or Version(self.version) < "0.5.0":
+            del self.options.with_unwind
 
     def configure(self):
         if self.options.shared:
             del self.options.fPIC
         if self.options.with_gflags:
             self.options["gflags"].shared = self.options.shared
 
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
     def requirements(self):
         if self.options.with_gflags:
             self.requires("gflags/2.2.2")
+        # 0.4.0 requires libunwind unconditionally
+        if self.options.get_safe("with_unwind") or (Version(self.version) < "0.5.0" and self.settings.os in ["Linux", "FreeBSD"]):
+            self.requires("libunwind/1.6.2")
```

Comment:
It's worth noting that it will break for users with a conan version < 1.51.1 and consuming static glog with a legacy generators, due to usage of layout (https://github.com/conan-io/conan/pull/11790). Might be a good idea to enforce these requires in package_info() since required_conan_version is 1.47.0, not 1.51.1 or higher.

Reply:
we can also bump the required version